markets/perps-market/test/integration/Account/ModifyCollateral.withdraw.test.ts

```diff
@@ -114,4 +117,155 @@ describe('ModifyCollateral Withdraw', () => {
+          liquidationRewardRatio: bn(0.05),
+        },
+        settlementStrategy: {
+          settlementReward: bn(0),
+        },
+      },
+    ];
+    const traderAccountIds = [2, 3];
+    const trader1AccountId = traderAccountIds[0];
+    const { systems, provider, trader1, perpsMarkets } = bootstrapMarkets({
+      synthMarkets: [],
+      perpsMarkets: perpsMarketConfigs,
+      traderAccountIds,
+    });
+    before('add collateral to margin', async () => {
+      await systems()
+        .PerpsMarket.connect(trader1())
+        .modifyCollateral(trader1AccountId, sUSDSynthId, bn(20_000));
+    });
+    let commonOpenPositionProps: Pick<
+      OpenPositionData,
+      'systems' | 'provider' | 'trader' | 'accountId' | 'keeper'
+    >;
+    before('identify common props', async () => {
+      commonOpenPositionProps = {
+        systems,
+        provider,
+        trader: trader1(),
+        accountId: 2,
+        keeper: trader1(),
+      };
+    });
+    before('open positions', async () => {
+      const positionSizes = [
+        bn(-2), // btc short
+        bn(20), // eth long
+      ];
+
+      for (const [i, perpsMarket] of perpsMarkets().entries()) {
+        await openPosition({
+          ...commonOpenPositionProps,
+          marketId: perpsMarket.marketId(),
+          sizeDelta: positionSizes[i],
+          settlementStrategyId: perpsMarket.strategyId(),
+          price: perpsMarketConfigs[i].price,
+        });
+      }
+    });
+    describe('account check after initial positions open', async () => {
+      it('should have correct open interest', async () => {
+        const expectedOi = 100_000; // abs((-2 * 30000) + (20 * 2000))
+        assertBn.equal(
+          await systems().PerpsMarket.totalAccountOpenInterest(trader1AccountId),
+          bn(expectedOi)
+        );
+      });
+      it('has correct pnl, given our position changed the skew', async () => {
+        const [btcPnl] = await systems().PerpsMarket.getOpenPosition(
+          trader1AccountId,
+          perpsMarkets()[0].marketId()
+        );
+        assertBn.equal(btcPnl, bn(-600));
+        const [ethPnl] = await systems().PerpsMarket.getOpenPosition(
+          trader1AccountId,
+          perpsMarkets()[1].marketId()
+        );
+        assertBn.equal(ethPnl, bn(-400));
+      });
+      it('has correct available margin', async () => {
+        assertBn.equal(
+          await systems().PerpsMarket.getAvailableMargin(trader1AccountId),
+          bn(19000) // collateral value  + pnl =  20000 + -1000
+        );
+      });
+    });
+    describe('withdraw with open position', () => {
+      it('allow withdraw when its less than "collateral available for withdraw"', async () => {
+        before('withdraw allowed amount', async () => {
+          await systems()
+            .PerpsMarket.connect(trader1())
+            .modifyCollateral(trader1AccountId, sUSDSynthId, bn(-17000));
+        });
```

Comment:
this is odd; you have the `before` inside the `it` block.  surprised the test passed

Reply:
Opsie, will fix

Reply:
Fixed!